src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,284 @@
+
+    task_list_file = os.path.join(job_dir, 'tasks.txt')
+    with open(task_list_file, "w") as f:
+        f.write(task_list)
+
+    return task_list_file, len(genome_ids_chunks)
+
+
+def _create_batch_script(job_dir, task_list_file, n_jobs):
+    """
+    Create the batch script (submit_taskfarmer.sl) for job submission
+    """
+    batch_script = '''#!/bin/sh\n'''
+    batch_script += f'''#SBATCH -N {n_jobs + 1} -c 64\n'''
+    batch_script += '''#SBATCH -q regular\n'''
+    batch_script += '''#SBATCH --time=4:00:00\n'''
+    batch_script += '''#SBATCH --time-min=0:30:00\n'''
+    batch_script += '''#SBATCH -C cpu\n\n'''
+    batch_script += f'''cd {job_dir}\n'''
+    batch_script += '''export THREADS=32\n\n'''
+    batch_script += f'''runcommands.sh {task_list_file}'''
+
+    batch_script_file = os.path.join(job_dir, 'submit_taskfarmer.sl')
+    with open(batch_script_file, "w") as f:
+        f.write(batch_script)
+
+    return batch_script_file
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Create the required documents/scripts for the TaskFarmer Workflow Manager'
+                    'and provide the option to execute tasks.')
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag arguments
+    required.add_argument('--tool', required=True, type=str, choices=TOOLS_AVAILABLE,
+                          help='Name of tool to be executed. (e.g. gtdb_tk, checkm2, etc.)')
+
+    required.add_argument(f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.KBASE_COLLECTION_DESCR)
+
+    required.add_argument(f'--{loader_common_names.LOAD_VER_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.LOAD_VER_DESCR)
+
+    required.add_argument('--source_data_dir', required=True, type=str,
+                          help='Source data (genome files) directory. '
+                               '(e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207')
+
+    # Optional arguments
+    optional.add_argument('--root_dir', type=str, default=loader_common_names.ROOT_DIR,
+                          help=f'Root directory for the collections project. (default: {loader_common_names.ROOT_DIR})')
+
+    optional.add_argument('--submit_job', action='store_true', help='Submit job to slurm')
+
+    args = parser.parse_args()
+
+    tool = args.tool
+    kbase_collection = getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME)
+    load_ver = getattr(args, loader_common_names.LOAD_VER_ARG_NAME)
+    source_data_dir = args.source_data_dir
+    root_dir = args.root_dir
+
+    current_datetime = datetime.datetime.now()
+    job_dir = os.path.join(root_dir, 'task_farmer_jobs', f'{tool}_{current_datetime.strftime("%Y_%m_%d_%H_%M_%S")}')
+    os.makedirs(job_dir, exist_ok=True)
+
+    _clone_repo(job_dir)
```

Comment:
I can't figure out why this is needed

Reply:
yea. I cannot remember why I did it. It shouldn't be needed. removed.

Reply:
:+1: